### PR TITLE
refextract: requests library usage and code style improvements

### PR DESCRIPTION
**Pull request by pedrogaudencio**
Still needs testing and reviews.

* Replaces urllib by requests.
* Saves the file locally using tempfile.
* Removes the file afterwards.
* Fixes indentation, imports and pep8 stuff.

### CODE REVIEW

**Comment by kaplun** [comment]
Hi! Note that the #2361 is still pending for integration. And is therefore not yet in pu :cry:

**Comment by jirikuncar** [comment]
@pedrogaudencio let's not wait for `legacy` PR. Please re-base and re-push. Thank you!

**Comment by pedrogaudencio** [comment]
@jirikuncar done ;)

**Comment by jirikuncar** [comment]
> Please don't forget to amend commit message too.

@pedrogaudencio please consult commit message format with @jalavik, @jmartinm or other colleague.

**Comment by jirikuncar** [comment]
Please check the code and commit message `kwalitee` locally before pushing to avoid Travis queue blocking. I have manually canceled test pushes.
image

**Comment by jirikuncar** [comment]
@pedrogaudencio can you please correctly format the commit message? Ask @jmartinm or @jalavik if you need help. Thank you!

--
You can also install `kwalitee` tool: `pip install --pre kwalitee` and run the checks locally: `kwalitee check message <sha or range like git log>`

File: invenio/legacy/refextract/config.py [hunk 1]
```diff
@@ -17,30 +17,35 @@
 # along with Invenio; if not, write to the Free Software Foundation, Inc.,
 # 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
 
-"""RefExtract configuration"""
+"""refextract configuration."""
+
+
+from invenio.base.config import CFG_VERSION
```
Comment (jirikuncar) [inline invenio/legacy/refextract/config.py]: It might be better to use `from invenio.version import version` instead.

**Comment by jirikuncar** [comment]
@pedrogaudencio what is the status of this PR?

--
cc @jmartinm @jalavik @kaplun

**Comment by jirikuncar** [comment]
@pedrogaudencio can you please follow the commit message format?

- empty line after header, between bullets, and before signatures

**Comment by pedrogaudencio** [comment]
Like so?

**Comment by jirikuncar** [comment]
For `* BETTER Replaces urllib by requests lib.` you can be more verbose as it's going to be included in RELEASE NOTES.

```
* BETTER Replaces usage of 'urllib' by 'requests' library and improves manipulation with temporary file used for extraction references.
```
(break line at maximal possible space before 72nd char)

add `Reviewed-by: me`

File: invenio/legacy/refextract/config.py [hunk 2]
```diff
@@ -17,32 +17,37 @@
...
 
 from __future__ import unicode_literals
```
Comment (jirikuncar) [inline invenio/legacy/refextract/config.py]: please make use that this is the **first** import in the file just after docstring.

File: invenio/legacy/refextract/api.py [hunk 3]
```diff
@@ -17,82 +17,95 @@
...
+"""This is where all the public API calls are accessible.
 
 This is the only file containing public calls and everything that is
 present here can be considered private by the invenio modules.
 """
 
 
-import os
-import sys
+from invenio.base.config import CFG_TMPSHAREDDIR
```
Comment (jirikuncar) [inline invenio/legacy/refextract/api.py]: You need to use either `from invenio.base.globals import cfg` or `from invenio.config import CFG_TMPSHAREDDIR`.

--
PS: Please re-order commits according to PEP8. Thank you!

**Comment by jirikuncar** [comment]
@pedrogaudencio it looks good now. Please ping me (on hangout) when you rebase. Thanks

**Pull request merged**